app/src/services/cpk.ts

```diff
@@ -280,6 +347,40 @@ class CPKService {
           to: collateral.address,
           value: marketData.funding,
         })
+      } else if (useCompoundReserve && compoundTokenDetails) {
+        const encodedMintFunction = CompoundService.encodeMintTokens(
+          compoundTokenDetails.symbol,
+          marketData.funding.toString(),
+        )
+        if (userInputCollateral.address === pseudoNativeAssetAddress) {
+          // If user chosen collateral is ETH
+          collateral = marketData.collateral
+          if (!this.cpk.isSafeApp()) {
+            txOptions.value = marketData.funding
+          }
+          transactions.push({
+            to: collateral.address,
+            data: encodedMintFunction,
+            value: marketData.funding.toString(),
+          })
+        } else {
+          collateral = marketData.collateral
+          // For any other compound pair that is not ETH
+          const encodedMintFunction = CompoundService.encodeMintTokens(
+            compoundTokenDetails.symbol,
+            marketData.funding.toString(),
+          )
+          // Approve cToken for the cpk contract
```

Comment:
I had trouble creating markets with DAI, I think we might need to add an extra step here: Transfer the underlying token from the user to the CPK before we mint the cToken.

Reply:
Yes, This was the case during an interim commit when I incorrectly resolved a merge conflict after rebasing to master. Should be fixed now :)